extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -142,6 +145,9 @@ class AmpCarousel extends AMP.BaseElement {
 
     const {element, win} = this;
     const slides = this.getRealChildren();
+    if (!slides || !slides.length) {
+      dev().warn(TAG, 'No slides were found.');
+    }
```

Comment:
Since the error tagged in #31898 occurs in shared files (`carousel.js` and `dimensions.js`), let's either add this same warning to all the components that use it (`amp-base-carousel`, `amp-inline-gallery`, `amp-stream-gallery`), or move the error to the shared file if possible - I think `carousel.js` would be preferred in that case. 

Another option could be to identify and protect all calls to `getBoundingClientRect` and `getDimension` (which is the one calling `getBoundingClientRect`) to make sure `undefined` is sensibly escaped such that it does not get passed there.

Reply:
Yes, I think moving it to carousel.js would be a good idea!